src/api/schemas/activity-schema.ts

```diff
@@ -87,5 +87,36 @@ const getFeeds = object({
     }).strict('Query contains invalid key')
 });
 
+const updateActivity = object({
+    // Params
+    params: object({
+        activityId: activityId
+    }).strict('Params contain invalid key'),
+    body: object({
+        update: object({
+            // Optional
+            title: title.optional(),
+            description: description.optional(),
+            address: address.optional(),
+            startTime: startTime.optional(),
+            endTime: endTime.optional(),
+            groupSize: groupSize.optional(),
+            costPerIndividual: costPerIndividual.optional(),
+            costPerGroup: costPerGroup.optional()
+        }).refine(
+            ({ title, description, address, startTime, endTime, groupSize, costPerIndividual, costPerGroup }) =>
+                title !== undefined ||
+                description !== undefined ||
+                address !== undefined ||
+                startTime !== undefined ||
+                endTime !== undefined ||
+                groupSize !== undefined ||
+                costPerIndividual !== undefined ||
+                costPerGroup !== undefined,
+            { message: 'One of the fields must be defined' }
```

Comment:
```suggestion
            (params) => Object.values(params).some((param) => param !== undefined),
            { message: 'One of the fields must be defined' }
```
I'm not sure whether `some` is the correct method or `every` would be better because I'm not 100% sure what you are trying to do but this will simplify this hard to read method for something simpler and easier to read. What do you think?